From c6bd5bb2ac83beccae1d5c043e6022049b3dbfd3 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Thu, 8 Nov 2012 18:32:51 -0800
Subject: [PATCH] gpio: xilinx_gpiops: Don't disable clocks if suspended

When entering suspend the device clocks were disabled, even
when they were already gated off due to RUNTIME_PM. Triggering
a kernel warning.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Acked: Michal Simek <michal.simek@xilinx.com>
---
 drivers/gpio/gpio-xilinxps.c |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-xilinxps.c b/drivers/gpio/gpio-xilinxps.c
index b89739d..5849500 100644
--- a/drivers/gpio/gpio-xilinxps.c
+++ b/drivers/gpio/gpio-xilinxps.c
@@ -439,7 +439,8 @@ static int xgpiops_suspend(struct device *_dev)
 			struct platform_device, dev);
 	struct xgpiops *gpio = platform_get_drvdata(pdev);
 
-	clk_disable(gpio->clk);
+	if (!pm_runtime_suspended(_dev))
+		clk_disable(gpio->clk);
 	return 0;
 }
 
@@ -449,7 +450,10 @@ static int xgpiops_resume(struct device *_dev)
 			struct platform_device, dev);
 	struct xgpiops *gpio = platform_get_drvdata(pdev);
 
-	return clk_enable(gpio->clk);
+	if (!pm_runtime_suspended(_dev))
+		return clk_enable(gpio->clk);
+
+	return 0;
 }
 
 #ifdef CONFIG_PM_RUNTIME
-- 
1.7.1

